storybookjs/storybook
# CLI: Warn the user for duplicated versions after automigrate

Closes #



## What I did

When users migrate projects, they might end up having issues in Storybook due to duplicated versions of dependencies, which might clash in the major range. They could come as direct but also transitive dependencies. This PR adds a feature to the CLI's `automigrate` command to warn users when such scenario happens.

Here's an example **when there are duplicated versions for conflicting packages** in a pnpm project:

<img width="663" alt="image" src="https://user-images.githubusercontent.com/1671563/228137039-74afe095-0383-43e6-a5e2-c3fdb48bed15.png">

Here's an example **when there are also duplicated major versions** in a yarn2 project:
<img width="756" alt="image" src="https://user-images.githubusercontent.com/1671563/228138529-b3601db2-5b79-4118-8489-b07716d2bf28.png">


## How to test

1. Create a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Install conflicting versions e.g. (yarn add @storybook/addon-actions@6.0.0)
3. Run `code/lib/cli/bin/index.js automigrate` in that sandbox
4. See the warning

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
